core/controllers/feedback.py

```diff
@@ -241,10 +240,11 @@ def post(self):
             if len(message_text) > 200:
                 message_text = message_text[:200] + '...'
 
-            if exploration.title in messages:
-                messages[exploration.title].append(message_text)
+            reference_dict = {exploration.title: message_text}
+            if exploration.id in messages:
+                messages[exploration.id].update(reference_dict)
```

Comment:
Also, do you have tests that submit multiple messages for the same exploration? They should have caught this...